app/src/main/java/com/artemchep/basics_multithreading/MainActivity.java

```
@@ -44,27 +46,41 @@ private void showWelcomeDialog() {
                 .show();
     }
 
-    public void onPushBtnClick(View view) {
+
+    public void onPushBtnClick(View view) throws InterruptedException {
         Message message = Message.generate();
-        insert(new WithMillis<>(message));
+        long startTime = SystemClock.currentThreadTimeMillis();
+        Log.d(TAG, "Current Time: " + startTime);
+        insert(new WithMillis<>(message, startTime));
     }
 
     @UiThread
-    public void insert(final WithMillis<Message> message) {
+    public void insert(final WithMillis<Message> message) throws InterruptedException {
```

Why does it need to throw an exception?

---

maybe I used queue.put() inside of addItemToQueue(). I replaced it to add().